src/hooks/useKeyboardState/index.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+
+import { KeyboardEvents } from "../../bindings";
+import { KeyboardController } from "../../module";
+
+// TODO: expose isVisible, isTransitioning (isHiding/isShowing) props?
+export const useKeyboardState = () => {
+  const [state, setState] = useState(KeyboardController.state());
+
+  useEffect(() => {
+    const didShowSubscription = KeyboardEvents.addListener(
+      "keyboardDidShow",
+      (e) => {
+        setState(e);
+      },
+    );
+    const didHideSubscription = KeyboardEvents.addListener(
+      "keyboardDidHide",
+      (e) => setState(e),
+    );
```

Comment:
so, you can create an array with handled events outside of hook.

```
const EVENTS = ['keyboardDidShow', 'keyboardDidHide'];
```

then in useEffect you can use it like that:

```
useEffect(() => {
   const subscribers = keyboardDidHide.map((event) => KeyboardEvents.addListener(event, setState))

   return () => subscribers.forEach((subscriber) => subscriber.remove())
}, [])
```